### DISCUSSION TITLE
Outflow boundary / outlet

### DISCUSSION THREAD

**Discussion by fabiocruz** [InteractiveComputerGraphics/SPlisHSPlasH#217] [Q&A] [answered]
Hello!

I am interested in using SPlisHSPlasH to simulate the fluid flow within a container with an initial fluid block, an inlet and an outlet.

I managed to mimic an inlet with an emitter, but I am having problems to setup an outlet. I thought this could be done by reusing particles that leave a given box (i.e. using the emitter parameters `emitterBoxMin` and `emitterBoxMax`), but this doesn't seem to remove particles leaving the box that are initially part of the fluid block. In other words, particles belonging to the emitter are removed when leaving the box defined by `emitterBoxMin` and `emitterBoxMax`, but fluid block particles are not. In my setup, both fluids (block and emitter) have the same properties (and thus the same material in SPlisHSPlasH), so it is critical that both particle sets are treated the same way.

Any suggestions on how to proceed? Thanks in advance!

**Comment by janbender** [comment, marked as answer]
The emitter should also remove particles from the fluid block if they have the same material. Here is an example

[Emitter_2D_2.zip](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/10174839/Emitter_2D_2.zip)

Note that the emitter only removes the amount of particles that it can reuse. The other particles which are not in the box and not reused yet are slowed down so that they do not disturb the simulation.

**Reply by fabiocruz** [reply]
Hi @janbender! Thanks for the answer and for the example input file. I'll give it a try and give feedback soon.

**Reply by fabiocruz** [reply]
@janbender, it worked perfectly. Thank you again for your quick feedback.
